torch/distributed/tensor/_api.py

```diff
@@ -346,6 +356,9 @@ def __coerce_same_metadata_as_tangent__(self, flatten_spec, expected_type=None):
     # pyre-fixme[3]: Return type must be annotated.
     # pyre-fixme[2]: Parameter must be annotated.
     def __torch_dispatch__(cls, func, types, args=(), kwargs=None):  # type: ignore[override]
+        fastpath_out = _maybe_fastpath_torch_dispatch(func, args, kwargs or {})
```

Comment:
Instead of doing a function call, you might consider just doing the condition inline and save a few more time